homeassistant/components/eight_sleep/sensor.py

```diff
@@ -53,37 +62,50 @@
 EIGHT_HEAT_SENSORS = ["bed_state"]
 EIGHT_ROOM_SENSORS = ["room_temperature"]
 
+VALID_TARGET_HEAT = vol.All(vol.Coerce(int), vol.Clamp(min=-100, max=100))
+VALID_DURATION = vol.All(vol.Coerce(int), vol.Clamp(min=0, max=28800))
 
-async def async_setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+SERVICE_EIGHT_SCHEMA = {
+    ATTR_TARGET: VALID_TARGET_HEAT,
+    ATTR_DURATION: VALID_DURATION,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: ep.AddEntitiesCallback
 ) -> None:
     """Set up the eight sleep sensors."""
-    if discovery_info is None:
-        return
-
-    eight: EightSleep = hass.data[DOMAIN][DATA_API]
-    heat_coordinator: DataUpdateCoordinator = hass.data[DOMAIN][DATA_HEAT]
-    user_coordinator: DataUpdateCoordinator = hass.data[DOMAIN][DATA_USER]
+    eight: EightSleep = hass.data[DOMAIN][entry.entry_id][DATA_API]
```

Comment:
It would be cleaner to do

`data = hass.data[DOMAIN][entry.entry_id]` and then use it below so you don't have all these hash lookups